test/basics.jl

```diff
@@ -0,0 +1,45 @@
+string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UniStr]
+
+@testset "constructors" begin
+    for strtype in string_types
+        @test convert(strtype, [0x61,0x62,0x63,0x21]) == "abc!"
+        @test convert(strtype, "abc!") == "abc!"
+
+        @test isempty(string())
+        @test eltype(GenericString) == Char
+        @test start("abc") == 1
```

Comment:
The following ones using "ab" and "abc" should all be changed to use values converted to the string type being tested.
`str_ab = convert(T, "ab")`
`str_abc = convert(T, "abc")`